x/orderbook/bucket_test.go

```diff
@@ -0,0 +1,138 @@
+package orderbook
+
+import (
+	"time"
+	"testing"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/coin"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+	"github.com/iov-one/weave/weavetest"
+)
+
+func TestMarketIDindexer(t *testing.T) {
+	cases := map[string]struct {
+		obj orm.Object
+		wantErr *errors.Error 
+	}{
+		"success": {
+			obj: orm.NewSimpleObj(nil, new(OrderBook)), 
+			wantErr: nil,
+		},
+		"not orderbook": {
+			obj: orm.NewSimpleObj(nil, new(Order)),
+			wantErr: errors.ErrState,
+		},
+	}
+
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			if _, err := marketIDindexer(tc.obj); !tc.wantErr.Is(err) {
+				t.Fatalf("unexpected error: %+v", err)
+			}
+		})
+	}
+}
+
+func TestOpenOrderIndexer(t *testing.T) {
+	cases := map[string]struct {
+		obj orm.Object
+		wantErr *errors.Error 
+	}{
+		"success": {
+			obj: orm.NewSimpleObj(nil, new(Order)), 
+			wantErr: nil,
+		},
+		//TODO BuildOpenOrderIndex cases 
+		"failure not order": {
+			obj: orm.NewSimpleObj(nil, new(OrderBook)),
+			wantErr: errors.ErrState,
+		},
+	}
+
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			if _, err := openOrderIndexer(tc.obj); !tc.wantErr.Is(err) {
+				t.Fatalf("unexpected error: %+v", err)
+			}
+		})
+	}
+}
+
+func TestOrderIDindexer(t *testing.T) {
+	cases := map[string]struct {
+		obj orm.Object
+		wantErr *errors.Error 
+	}{
+		"success": {
+			obj: orm.NewSimpleObj(nil, new(Trade)), 
+			wantErr: nil,
+		},
+		"not trade": {
+			obj: orm.NewSimpleObj(nil, new(OrderBook)),
+			wantErr: errors.ErrState,
+		},
+	}
+
+	for testName, tc := range cases {
+		t.Run(testName, func(t *testing.T) {
+			if _, err := orderIDIndexer(tc.obj); !tc.wantErr.Is(err) {
+				t.Fatalf("unexpected error: %+v", err)
+			}
+		})
+	}
+}
+
+func TestOrderBookTimedIndexer(t *testing.T) {
+	now := weave.AsUnixTime(time.Now())
+	invalidTime:= weave.AsUnixTime(time.Unix(-1, 0))
+
+	validTrade := &Trade{
```

Comment:
I like the actual fixtures with real data (non-zero values).
It should be pretty easy to assert what the return value is in the success case